### PR TITLE
Add bottom menu when shake for support

### LINKED ISSUE #395: Improve "Shake for support" experience
Users trigger contact support by shaking their device when they don't expect that to happen. It is friction in the user's experience and creates meaningless support tickets.

More context in thread: https://celo-org.slack.com/archives/C018VLA3YAK/p1619540156313900

Proposed solution: show a modal that will explain what had happened and will allow the user to contact support or go back to the screen where they have triggered it. Design in Figma: https://www.figma.com/file/mlLC6CfvtEgcpX91gmfHFR/Shake-to-Contact?node-id=1%3A3

### PR DESCRIPTION
**Pull request by isabellewei**
### Description

When the user shakes their phone, a bottom menu will appear giving them the option to progress to the Contact Support screen

### Tested

Screenshot_20210602-111651_Alfajores (dev)

### How others should test

Shake phone from any screen in the app

### Related issues

- Fixes #395 

### Backwards compatibility
Yes

### CODE REVIEW

File: packages/mobile/src/navigator/NavigatorWrapper.tsx [hunk 1]
```diff
@@ -200,6 +216,26 @@ export const NavigatorWrapper = () => {
...
+            <Text style={styles.supportTitle}>{i18n.t('accountScreen10:havingTrouble')}</Text>
+            <Text style={styles.supportSubtitle}>{i18n.t('accountScreen10:shakeForSupport')}</Text>
+            <Button
+              onPress={onContactSupport}
+              text={i18n.t('accountScreen10:contactSupport')}
+              type={BtnTypes.PRIMARY}
+              testID="ContactSupportFromShake"
+            />
+          </View>
+        </Modal>
```
Comment (jeanregisser) [inline packages/mobile/src/navigator/NavigatorWrapper.tsx:238]: Would be nice to extract this into its own component and add a simple unit test for it (checking tap triggers the right action?).

File: packages/mobile/src/navigator/NavigatorWrapper.tsx [hunk 2]
```diff
@@ -225,6 +261,33 @@ const styles = StyleSheet.create({
     right: 0,
     bottom: 0,
   },
+  modal: {
+    margin: 0,
+  },
+  shakeForSupport: {
+    position: 'absolute',
+    bottom: 0,
+    height: 240,
```
Comment (jeanregisser) [inline packages/mobile/src/navigator/NavigatorWrapper.tsx:270]: We should avoid hardcoding the `height`, and let the content define it, so there's less risk that the layout breaks on devices with a narrow screen, or if the font size changes.

File: packages/mobile/src/navigator/NavigatorWrapper.tsx [hunk 3]
```diff
@@ -200,6 +216,26 @@ export const NavigatorWrapper = () => {
           <AlertBanner />
           <InviteFriendModal isVisible={isInviteModalVisible} onInvite={onInvite} />
         </View>
+        <Modal isVisible={shakeForSupportVisible} backdropOpacity={0.5} style={styles.modal}>
+          <View style={styles.shakeForSupport}>
```
Comment (jeanregisser) [inline packages/mobile/src/navigator/NavigatorWrapper.tsx:220]: We should wrap this in `<SafeAreaView edges={['bottom']}>` so it doesn't collide with (or stay too close to) the home indicator on iOS devices.

**Review by jeanregisser — COMMENTED** [review]
Looking good! 👍

File: packages/mobile/src/account/ShakeForSupport.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          text={i18n.t('accountScreen10:contactSupport')}
+          type={BtnTypes.PRIMARY}
+          testID="ContactSupportFromShake"
+        />
+      </SafeAreaView>
+    </Modal>
+  )
+}
+
+const styles = StyleSheet.create({
```
Comment (MuckT) [inline packages/mobile/src/account/ShakeForSupport.tsx:72]: It looks like the support banner isn't centered in all devices: [iPhone 11](https://drive.google.com/file/d/1EUVCWuYh32HR1lqVFpfQJJZTmg0gA426/view?usp=sharing)

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R37](https://valora.testquality.com/project/12128/testcase/286489/requirement/11464).

File: packages/mobile/src/account/ShakeForSupport.tsx (hunk 4 above)
Comment (isabellewei) [reply packages/mobile/src/account/ShakeForSupport.tsx:72]: thanks for catching that!

**Review by jeanregisser — APPROVED** [review]

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R61](https://valora.testquality.com/project/12128/testcase/292788/requirement/11555).

**Comment by Celoqa** [comment]
Hi @MuckT We have verified this issue on Latest android build V1.16.0 & Testflight build V1.16.0(66) and updated the status in testqulaity tool.

**Testquality tool link:**

**Android:**
- https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432159 

**iOS:**
- https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432187

**New issue:**
- https://github.com/celo-org/wallet/issues/849

**Comment by Celoqa** [comment]
Hi @MuckT, We have verified this issue on Latest android build V1.16.0 in Android 11.0 OS and everything is working as expected.
**Device Details:** Google Pixel 4a(11.0)

**Pull request merged**